File: PDOO-Civitas/CivitasR/lib/sorpresa_carcel.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Sorpresa_carcel < Sorpresa
    
    public
    def initialize(tablero)
      
      @tablero = tablero
      @mazo = Mazo_Sorpresas.new
      @texto = "Esta sorpresa te lleva a la cárcel"
      
    end
    
    public
    def aplicar_a_jugador(actual, todos)
      
      if(jugador_correcto(actual, todos))
        
        informe(actual, todos)
        todos.at(actual).encarcelar(@tablero.num_casilla_carcel)
      end
      
    end
    
    private
    def informe(actual, todos)
      
      Diario.instance.ocurre_evento("Se esta aplicando una sorpresa IR A CÁRCEL al jugador " + todos.at(actual).to_s);
      
    end
    
    public
    def to_s()
      
      return "Tipo: Sorpresa Cárcel\n" + @texto

    end
  
    public_class_method :new

  end
end

File: PDOO-Civitas/CivitasJ/src/civitas/SorpresaCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ramonjtorres
 */
public class SorpresaCarcel extends Sorpresa{
    
    private String texto;
    MazoSorpresas mazo;
    Tablero tablero;
    
    SorpresaCarcel(Tablero tablero){
    
        this.tablero = tablero;
        this.mazo = new MazoSorpresas();
        this.texto = "Esta sorpresa te lleva a la cárcel";
        this.informacion = this.texto;
    
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
    
        if(this.jugadorCorrecto(actual, todos)){
        
            this.informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
        
    }
    
    @Override
     void informe(int actual, ArrayList<Jugador> todos){
    
        Diario.getInstance().ocurreEvento("Se esta aplicando una sorpresa IR A CÄRCEL al jugador " + todos.get(actual).getNombre());
    }
    
    @Override
    public String toString(){
        
        String sorpresa = "Tipo: Sorpresa Cárcel\n";
        
        sorpresa += this.texto;
        
        return sorpresa;
    }
    
}
